src/main/java/FigureSupplier.java

```diff
@@ -0,0 +1,37 @@
+import java.util.Random;
+
+public class FigureSupplier {
+    public static final int AMOUNT_OF_FIGURES = 5;
+    public static final int BORDER = 5;
+    private Random random = new Random();
+
+    public Figure getRandomFigure() {
+        int randomNumber = random.nextInt(AMOUNT_OF_FIGURES - 1);
+
+        switch (randomNumber) {
+            case 1:
+                return new Square(new Random().nextInt(BORDER - 1) + 1,
+                        new ColorSupplier().getRandomColor());
```

Comment:
mate-academy.github.io/jv-program-common-mistakes/java-core/abstract-class-interface/oop-advanced#%D1%81reating-a-figure-dont-pass-expressions-in-constructor

we can create separate methods to create random figures, for example, getRandomSquare, inside this method you should create random variables that you will pass to the constructor and then return triangle with random values. And then you can use these methods inside your switch/case